dask_planner/src/expression.rs

```diff
@@ -821,10 +778,27 @@ impl PyExpr {
     }
 }
 
+impl PyExpr {
+    /// Get the scalar value represented by this literal expression, returning an error
+    /// if this is not a literal expression
+    fn get_scalar_value(&self) -> Result<&ScalarValue> {
+        match &self.expr {
+            Expr::Literal(v) => Ok(v),
+            _ => Err(DaskPlannerError::Internal(
+                "get_scalar_value() called on non-literal expression".to_string(),
+            )),
+        }
+    }
+}
+
+fn unexpected_literal_value() -> PyErr {
+    DaskPlannerError::Internal("getValue<T>() - Unexpected value".to_string()).into()
+}
```

Comment:
Would it make sense to display the invalid scalar type received as part of this error message?